### PR TITLE
Small improvements to docstyle and typing in util functions

### PR DESCRIPTION
**Pull request by omar-rifai**
Moving previously copied legacy code in #725 back to it's original place with a few enhancements. 

The aim is to remove the duplicated code in #725 and reduce the scope of the PR to only Pydra changes.

### CODE REVIEW

File: clinica/iotools/utils/data_handling.py [hunk 1]
```diff
@@ -1086,71 +1101,79 @@ def check_volume_location_in_world_coordinate_system(
...
-        nii_volume: path to nii volume
-
-    Returns:
-        [Returns]
+def get_world_coordinate_of_center(nii_volume: PathLike) -> ndarray:
+    """Extract the world coordinates of the center of the image.
+    Parameters
+    ---------
+    nii_volume : PathLike
+        path to nii volume
```
Comment (NicolasGensollen) [inline clinica/iotools/utils/data_handling.py:1151]:
```suggestion
        path to nii volume
        
```

File: clinica/iotools/utils/data_handling.py [hunk 2]
```diff
@@ -1086,71 +1101,79 @@ def check_volume_location_in_world_coordinate_system(
...
+    Returns
+    -------
+    tuple
```
Comment (NicolasGensollen) [inline clinica/iotools/utils/data_handling.py:1154]:
```suggestion
    np.ndarray :
```

File: clinica/iotools/utils/data_handling.py [hunk 3]
```diff
@@ -1086,71 +1101,79 @@ def check_volume_location_in_world_coordinate_system(
...
     import numpy as np
     from nibabel.filebasedimages import ImageFileError
 
+    from clinica.utils.exceptions import ClinicaException
     from clinica.utils.stream import cprint
 
-    assert isinstance(nii_volume, str), "input argument nii_volume must be a str"
-    assert isfile(nii_volume), "input argument must be a path to a file"
+    # from clinica.utils.stream import cprint
+    # assert isinstance(nii_volume, str), "input argument nii_volume must be a str"
```
Comment (NicolasGensollen) [inline clinica/iotools/utils/data_handling.py:1171]: to be removed ?

File: clinica/iotools/utils/data_handling.py [hunk 4]
```diff
@@ -1189,34 +1212,44 @@ def get_world_coordinate_of_center(nii_volume):
     return center_coordinates_world
 
 
-def get_center_volume(header):
+def get_center_volume(header: dict) -> ndarray:
```
Comment (NicolasGensollen) [inline clinica/iotools/utils/data_handling.py:1215]:
```suggestion
def get_center_volume(header: Nifti1Header) -> ndarray:
```

File: clinica/iotools/utils/data_handling.py [hunk 5]
```diff
@@ -1189,34 +1212,44 @@ def get_world_coordinate_of_center(nii_volume):
...
     """Get the voxel coordinates of the center of the data, using header information.
-
-    Args:
-        header: a nifti header
-
-    Returns:
+    Parameters
+    ----------
+        header: Nifti1Header
```
Comment (NicolasGensollen) [inline clinica/iotools/utils/data_handling.py:1219]:
```suggestion
    header: Nifti1Header
```

File: clinica/iotools/utils/data_handling.py [hunk 6]
```diff
@@ -1339,15 +1383,21 @@ def get_aff_matrix(h):
...
-def vox_to_world_space_method_3_bis(coordinates_vol, header):
+def vox_to_world_space_method_3_bis(coordinates_vol: ndarray, header: Nifti1Header):
     """
-    This method relies on the same technique as method 3, but for images created by FreeSurfer (MGHImage, MGHHeader).
-    Args:
-        coordinates_vol: coordinate in the volume (raw data)
-        header: nib.freesurfer.mghformat.MGHHeader object
-
-    Returns:
+    Parameters
```
Comment (NicolasGensollen) [inline clinica/iotools/utils/data_handling.py:1388]:
```suggestion
    Convert coordinates to world space (method 3 bis).
    
    Parameters
```

File: clinica/pipelines/t1_volume_tissue_segmentation/t1_volume_tissue_segmentation_utils.py [hunk 7]
```diff
@@ -42,29 +43,35 @@ def zip_list_files(class_images, zip_files=False):
...
+    dartel_tissues: list
+        Classes of images to save for DARTEL template calculation. Ex: [1] is only GM'
+    save_warped_unmodulated : bool
+        Save warped unmodulated images for tissues specified in --tissue_classes
+    save_warped_modulated: bool
+        Save warped modulated images for tissues specified in --tissue_classes
+    Returns
+    -------
+    list
+        List of tuples according to NewSegment input por tissues
```
Comment (NicolasGensollen) [inline clinica/pipelines/t1_volume_tissue_segmentation/t1_volume_tissue_segmentation_utils.py:68]:
```suggestion
        List of tuples according to NewSegment input for tissues
```

File: clinica/utils/spm.py [hunk 8]
```diff
@@ -37,11 +39,12 @@ def check_spm_home():
...
 
-def get_tpm():
+def get_tpm() -> PathLike:
     """Get Tissue Probability Map (TPM) from SPM.
-
-    Returns:
-        str: TPM.nii from SPM
+    Returns
+    -------
+    str
```
Comment (NicolasGensollen) [inline clinica/utils/spm.py:46]:
```suggestion
    PathLike :
```

**Review by NicolasGensollen — COMMENTED** [review]
Thanks @omar-rifai ! Looking good ! 
Here are some suggestions, mostly on formatting.

File: clinica/iotools/utils/data_handling.py [hunk 9]
```diff
@@ -1086,71 +1101,79 @@ def check_volume_location_in_world_coordinate_system(
...
+
+def is_centered(nii_volume: PathLike, threshold_l2: int = 50) -> bool:
+    """Checks if a NIfTI volume is centered on the origin of the world coordinate system.
+    Parameters
+    ---------
+    nii_volume : PathLike
+        path to NIfTI volume
+    threshold_l2: int
+        Maximum distance between origin of the world coordinate system and the center of the volume to
+        be considered centered. The threshold were SPM segmentation stops working is around 100 mm (it was determined empirically after several trials on a generated dataset), so default value is 50mm in order to have a security margin, even when dealing with co-registered files afterward.
```
[the remark this answers is not in the document]
Comment (omar-rifai) [reply clinica/iotools/utils/data_handling.py:1126]: the new format looks a bit odd?

**Review by NicolasGensollen — APPROVED** [review]

**Pull request merged**